### ISSUE TITLE
memory leak in pred accumulation

### ISSUE THREAD

**Issue by potipot** [airctic/icevision#736]
## 🐛 Bug
Metric accumulates the entire pred object, together with the image itself resulting in a memory leak.

**To Reproduce**
Validate on any large dataset
**Expected behavior**
Images shouldn't be stored in memory for validation

**Comment by potipot** [comment]
The issue seems to be coming from having a too large batch size. The model and batch data fits in memory but during validation, when using num_workers>0, starts leaking on validation. Not sure if this is an icevision problem, or pytorch lightning problem. 
Investigating WIP

**Comment by potipot** [comment]
So the error is actually here:

```python
def convert_raw_predictions(
    raw_preds: torch.Tensor, records: Sequence[BaseRecord], detection_threshold: float
) -> List[Prediction]:
    dets = raw_preds.detach().cpu().numpy()
    preds = []
    for det, record in zip(dets, records):
        if detection_threshold > 0:
            scores = det[:, 4]
            keep = scores > detection_threshold
            det = det[keep]

        pred = BaseRecord(
            (
                ScoresRecordComponent(),
                ImageRecordComponent(),
                InstancesLabelsRecordComponent(),
                BBoxesRecordComponent(),
            )
        )
```

The `pred`  is a BaseRecord that uses `ImageRecordComponent` but not `FilepathRecordComponent`. When calling unload, the image persists in memory, as `ImageRecordComponent` itself doesn't override the `_unload` method. 

One solution would be to overload the `_unload` method for `ImageRecordComponent` the same way it is done for `FilepathRecordComponent`. 

Otherwise we might add the extra component in pred BaseRecord initialization. @lgvaz what do you think?

**Comment by lgvaz** [comment]
Good catch!! Yeah, it seems a reasonable idea to implement `_unload` for `ImageRecordComponent`, and because `FilepathRecordComponent` inherits from it we can remove the method from here right?

---
 This is also related to [this](https://github.com/airctic/icevision/issues/692), Calling unload on cocometric seems a bit hacky. The problem is that currently `convert_raw_predictions` is used both in inference (when we want to keep the image) and in validation (where we don't, at least for the majority of cases. But maybe if we are training gans or something similar we would like to keep the images?)
 
It's not clear how to cleanly do this :sweat_smile:

**Comment by potipot** [comment]
solved by https://github.com/airctic/icevision/pull/737
